boa/src/builtins/json/tests.rs

```diff
@@ -189,3 +189,49 @@ fn json_stringify_return_undefined() {
     assert_eq!(actual_function, expected);
     assert_eq!(actual_symbol, expected);
 }
+
+#[test]
+fn json_parse_array_with_reviver() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let result = forward(
+        &mut engine,
+        r#"JSON.parse('[1,2,3,4]', function(k, v){
+            if (typeof v == 'number') {
+                return v * 2;
+            } else {
+                v
+        }})"#,
+    );
+    assert_eq!(
+        result,
+        "{\n    3: 8,\n    2: 6,\n    0: 2,\n    1: 4,\n    length: 4,\n    extensible: true\n}"
+    );
```

Comment:
I think a better way of doing this is to call `forward_val` which returns a `Value` and call `.get_field` and then call `to_number()` on them to get a number that you can assert against

Reply:
Done